app/models/user/identifiable.rb

```diff
@@ -0,0 +1,8 @@
+module User::Identifiable
+  extend ActiveSupport::Concern
+
+  included do
+    validates :email_address, presence: true, uniqueness: true, format: {with: URI::MailTo::EMAIL_REGEXP}
+    encrypts :email_address, downcase: true, deterministic: true
```

Comment:
I know we had a brief chat about testing out ActiveRecord::Encryption on `email_address` and `name`. Im realizing that these encrypted fields may make it difficult to work with https://github.com/ankane/blazer. Blazer does have a way to [transform variables](https://github.com/ankane/blazer#encrypted-data), but I'm not sure if it works with ActiveRecord::Encryption.

What are your thoughts on this? I'd really love to be able to use Blazer

Reply:
Touching on Active Record Encryption first, I would be weary of any analytic query that dealt with sensitive attributes like emails or names. One of the primary motivations behind the creation of ARE isn't just an extra layer of security (cause there's not much you can do anyway if someone manages to get ahold of an SSH key for production servers), but - perhaps more importantly - [to define what constitutes user data at the application level](https://world.hey.com/jorge/privacy-aware-rails-consoles-with-console1984-and-audits1984-53562eaa), which allows for tools like [console1984](https://github.com/basecamp/console1984) and automatic parameter redaction.

Regarding Blazer, I think we should avoid it unless there's a strong need or benefit to it that outweighs the risks and disadvantages. One of the nice things about Rails is that you can be somewhat [naive with things](https://rubyonrails.org/doctrine#no-one-paradigm) like mastering SQL, especially in the beginning - now I'm not saying you shouldn't craft [a few raw fragments of SQL when truly needed](https://dev.37signals.com/active-record-nice-and-blended/#:~:text=while%20this%20is%20a%20bit%20of%20an%20edge%20case), but Blazer sits outside the majestic world of Ruby, the fresh air that Rails abstractions provide, and executes raw queries straight from the UI which doesn't sit well with me. As expressed in the first link, one of the main ways `console1984` + `audits1984` is used is for investigating and identifying needs for admin tools, which can then be implemented in a deliberate and likely more thoughtful manner while also being tested, reviewed, versioned, and tracked along with the rest of the application.